soh/src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -209,12 +216,82 @@ void func_809DF730(EnCow* this, GlobalContext* globalCtx) {
     }
 }
 
+int EnCow_GetCowId(EnCow* this, GlobalContext* globalCtx) {
+    s32 uniqueCoords = this->actor.world.pos.x + this->actor.world.pos.z;
+
+    switch (globalCtx->sceneNum) {
+        case SCENE_SOUKO: // Lon Lon Tower
+            switch (uniqueCoords) {
+                // Two cases here cause this cow is moved in randomizer
+                case -173:
+                case -72:
+                    return 0;
+                default:
+                    return 1;
+            }
+        case SCENE_MALON_STABLE:
+            switch (uniqueCoords) {
+                // Two cases here cause this cow is moved in randomizer
+                case -257:
+                case -138:
+                    return 2;
+                default:
+                    return 3;
+            }
+        case SCENE_KAKUSIANA: // Grotto
+            switch (uniqueCoords) {
+                case 1973:
+                    return 4;
+                default:
+                    return 5;
+            }
+        case SCENE_LINK_HOME:
+            return 6;
+        case SCENE_LABO: // Impa's house
+            return 7;
+        case SCENE_SPOT09: // Gerudo Valley
+            return 8;
+        // TODO: Handle Jabu MQ Cow
+    }
+}
+
+void EnCow_MoveCowsForRandomizer(EnCow* this, GlobalContext* globalCtx) {
+    int cowId = EnCow_GetCowId(this, globalCtx);
+
+    // Move left cow in lon lon tower
+    if (cowId == 0) {
+        this->actor.world.pos.x = -229.0f;
+        this->actor.world.pos.z = 157.0f;
+        this->actor.shape.rot.y = 15783.0f;
+    // Move right cow in lon lon stable
+    } else if (cowId == 2) {
+        this->actor.world.pos.x += 119.0f;
+    }
+}
+
+GetItemID EnCow_GetRandomizerItemFromCow(EnCow* this, GlobalContext* globalCtx, bool setFlag) {
+    GetItemID itemId = ITEM_NONE;
+    int cowId = EnCow_GetCowId(this, globalCtx);
```

Comment:
with the previous change i recommended, instead of having a `cowId` here we'd have the `RandomizerCheck` (such as `RC_KF_LINKS_HOUSE_COW`)